### PR TITLE
indexable Spring application context

### PR DESCRIPTION
**Pull request by emilioOcaEns**

### CODE REVIEW

File: modules/fox-spring/src/main/java/com/ensolvers/fox/spring/lightweightcontainer/GlobalApplicationContext.java [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+ *      <p>
+ *      <p>
+ *      * Construction allows building on the fly new context by offering legacy configuration xml files. <p>
+ *      Example:  <code> MyComponent component = (MyComponent)GlobalApplicationContext.getBean( "MyComponent" );</code> <p>
+ *
+ *      <p>
+ *
+ */
+@Component
+public class GlobalApplicationContext implements ApplicationContextAware {
```
Comment (josematiasrivero) [inline modules/fox-spring/src/main/java/com/ensolvers/fox/spring/lightweightcontainer/GlobalApplicationContext.java:38]: Add documentation to all public methods following general standards of Java open source projects

File: modules/fox-spring/src/main/java/com/ensolvers/fox/spring/lightweightcontainer/GlobalApplicationContext.java [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+        }
+        return getInstance().getBean(name);
+    }
+
+    public static ApplicationContext getInstance() {
+        return context;
+    }
+
+    public static Collection<String> getContextsFromProperty( InputStream stream ) throws IOException {
+        Vector contexts = new Vector();
```
Comment (josematiasrivero) [inline modules/fox-spring/src/main/java/com/ensolvers/fox/spring/lightweightcontainer/GlobalApplicationContext.java:60]: Vector is a very old class. The only perceivable advantage that I'm aware of is that the method is that their methods are synchronized. Of course that this comes with a cost. Do we need it? Can't we use an efficient implementation provided by  java.util.concurrent.* ?

Comment (josematiasrivero) [reply modules/fox-spring/src/main/java/com/ensolvers/fox/spring/lightweightcontainer/GlobalApplicationContext.java:60]: Now, re-reading this, why not using Apache IOUtils / FileUtils methods?

File: modules/fox-spring/src/main/java/com/ensolvers/fox/spring/lightweightcontainer/GlobalApplicationContext.java [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+            while ( (line = reader.readLine() ) != null ) {
+                contexts.add( line );
+                Logger.debug( GlobalApplicationContext.class, "context line read [" + line + "]" );
+            }
+        }
+
+        return contexts;
+    }
+
+    public static Collection<String> getContextsFromProperty( String propertyClassPath ) throws IOException {
```
Comment (josematiasrivero) [inline modules/fox-spring/src/main/java/com/ensolvers/fox/spring/lightweightcontainer/GlobalApplicationContext.java:73]: propertyClassPath -> propertyClasspath

File: modules/fox-spring/src/main/java/com/ensolvers/fox/spring/lightweightcontainer/GlobalApplicationContext.java [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+        return getContextsFromProperty( stream );
+    }
+
+    public static String[] applyDevelopContextFilter(String[] paths) throws ApplicationContextException {
+        Vector<String> filtered = new Vector();
+        try {
+            for (String path : paths) {
+                Logger.info(GlobalApplicationContext.class, "configLocations [" + path + "]");
+                Resource[] resources = new PathMatchingResourcePatternResolver().getResources(path);
+                Logger.debug(GlobalApplicationContext.class, "resources obtenidos segun locations [" + resources.length + "]");
```
Comment (josematiasrivero) [inline modules/fox-spring/src/main/java/com/ensolvers/fox/spring/lightweightcontainer/GlobalApplicationContext.java:88]: Need to translate everything to english, in addition to document all these methods in english to

File: modules/fox-spring/src/main/java/com/ensolvers/fox/spring/lightweightcontainer/GlobalApplicationContext.java [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+                for (Resource resource : resources) {
+                    Logger.info(GlobalApplicationContext.class, "subLocation from [" + path + "], [" + resource + "]");
+                    if (resource.getFilename().toString().indexOf("develop") == -1) {
+                        /* *
+                         * ClassPathResource o FileSystemResource suported, both interfaces with getPath method
```
Comment (josematiasrivero) [inline modules/fox-spring/src/main/java/com/ensolvers/fox/spring/lightweightcontainer/GlobalApplicationContext.java:93]: suported -> supported

Also, why are we using multiline comments while we have a one-liner here?

File: modules/fox-spring/src/main/java/com/ensolvers/fox/spring/lightweightcontainer/GlobalApplicationContext.java [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+ *      <p>
+ *      * Look up allows to get from Spring context any registered bean  <p>
+ *      Example:  <code> MyComponent component = (MyComponent)GlobalApplicationContext.getBean( "MyComponent" );</code> <p>
+ *      <p>
+ *      <p>
+ *      * Construction allows building on the fly new context by offering legacy configuration xml files. <p>
+ *      Example:  <code> MyComponent component = (MyComponent)GlobalApplicationContext.getBean( "MyComponent" );</code> <p>
+ *
+ *      <p>
+ *
```
Comment (josematiasrivero) [inline modules/fox-spring/src/main/java/com/ensolvers/fox/spring/lightweightcontainer/GlobalApplicationContext.java:35]: Need to provide examples for other public methods as well

File: modules/fox-spring/src/test/java/com/ensolvers/fox/spring/AlternativeGlobalApplicationContextTest.java [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+
+public class AlternativeGlobalApplicationContextTest {
+
+   @Test
+   public void testResourceSample() {
+      GlobalApplicationContext.build( "classpath*:/com/ensolvers/fox/spring/sameAutowiring-context.xml");
+      SampleComponent sc = (SampleComponent)GlobalApplicationContext.getBean("sampleComponent" );
+      assertNotNull( sc );
+      assertEquals( "Hey there", sc.helloWorld() );
+
```
Comment (josematiasrivero) [inline modules/fox-spring/src/test/java/com/ensolvers/fox/spring/AlternativeGlobalApplicationContextTest.java:23]: Extra space

File: modules/fox-spring/src/test/resources/com/ensolvers/fox/spring/pointcutAutowiring-context.xml [hunk 8]
```diff
@@ -0,0 +1,11 @@
...
+<beans xmlns="http://www.springframework.org/schema/beans"
+   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+   xmlns:context="http://www.springframework.org/schema/context"
+   xsi:schemaLocation="http://www.springframework.org/schema/beans      http://www.springframework.org/schema/beans/spring-beans-3.0.xsd
+                       http://www.springframework.org/schema/context    http://www.springframework.org/schema/context/spring-context-3.0.xsd
+                       ">
+
+   <context:annotation-config />
+   <context:component-scan base-package="com.ensolvers.fox.spring.aop" />
+</beans>
```
Comment (josematiasrivero) [inline modules/fox-spring/src/test/resources/com/ensolvers/fox/spring/pointcutAutowiring-context.xml:11]: Why are we using XML here? We need to use annotation-based configurations as we do in other projects as well

**Review by josematiasrivero — CHANGES_REQUESTED** [review]

File: modules/fox-spring/src/test/java/com/ensolvers/fox/spring/AlternativeGlobalApplicationContextTest.java (hunk 7 above)
Comment (emilioOcaEns) [reply modules/fox-spring/src/test/java/com/ensolvers/fox/spring/AlternativeGlobalApplicationContextTest.java:23]: fixed

File: modules/fox-spring/src/test/resources/com/ensolvers/fox/spring/pointcutAutowiring-context.xml (hunk 8 above)
Comment (emilioOcaEns) [reply modules/fox-spring/src/test/resources/com/ensolvers/fox/spring/pointcutAutowiring-context.xml:11]: Annotated examples will come as we gain experience.
XML is the first configuration kind Spring searches and is useful to provide simple or complex isolated contexts

File: modules/fox-spring/src/main/java/com/ensolvers/fox/spring/lightweightcontainer/GlobalApplicationContext.java (hunk 1 above)
Comment (emilioOcaEns) [reply modules/fox-spring/src/main/java/com/ensolvers/fox/spring/lightweightcontainer/GlobalApplicationContext.java:54]: fixed

File: modules/fox-spring/src/main/java/com/ensolvers/fox/spring/lightweightcontainer/GlobalApplicationContext.java (hunk 2 above)
Comment (emilioOcaEns) [reply modules/fox-spring/src/main/java/com/ensolvers/fox/spring/lightweightcontainer/GlobalApplicationContext.java]: Most important feature is that Vector has better and more usable protocol. Been synchronized or concurrent is of no use here, neither performance has an impact. 
Why use third party libraries if old proven Objects work well enough and are more versatile

**Pull request merged**